src/drivers/net/virtio/mod.rs

```diff
@@ -352,14 +345,22 @@ impl NetworkDriver for VirtioNetDriver<Init> {
 			.unwrap();
 
 		for _ in 1..num_buffers {
-			let (header, packet) = self.receive_single_packet()?;
+			let mut buffer_tkn = self.inner.recv_vqs.get_next()?;
+			// The descriptor that was meant for the header of another frame was used for a portion of the current frame's contents.
+			// Thus, we cannot cast it to a Hdr.
+			let (header_descriptor, used_len) = buffer_tkn.used_recv_buff.pop_front_raw()?;
+			combined_packets.extend_from_slice(unsafe {
+				core::slice::from_raw_parts(Box::as_ptr(&header_descriptor).cast::<u8>(), used_len)
```

Comment:
As always, I'd like to avoid unavoidable nightly features (box_as_ptr):

```suggestion
				slice::from_raw_parts((&raw const *header_descriptor).cast::<u8>(), used_len)
```